INCode/models.py

```diff
@@ -165,41 +192,177 @@ def get_translation_unit(self):
     def is_definition(self):
         return self.cursor_.is_definition()
 
-    def get_referenced_callables(self):
-        return [self.index_.lookup(usr) for usr in self.referenced_usrs_]
+    @property
+    def referenced_callables(self):
+        return [Index().lookup(usr) for usr in self.referenced_usrs_]
+
+    @property
+    def caller(self):
+        return self.caller_
 
     def initialize(self):
-        for cursor in self.cursor_.walk_preorder():
-            if Callable._is_a_call(cursor) and self.index_.is_interesting(cursor):
-                definition = cursor.referenced
-                callable = Callable(definition, self.index_, False)
-                if callable.get_id() not in self.referenced_usrs_ and definition.kind != CursorKind.CONSTRUCTOR:
-                    self.referenced_usrs_.append(callable.get_id())
-                self.index_.register(callable)
+        self._load_cursors(self.cursor_.get_children())
+
+    def _load_cursors(self, cursors):
+        for cursor in cursors:
+            if cursor.kind != CursorKind.LAMBDA_EXPR:
+                self._load_cursors(cursor.get_children())
+            if Callable._is_a_call(cursor) and Index().is_interesting(cursor):
+                definition = cursor.referenced if cursor.referenced else cursor
+                if Caller.is_supported(definition):
+                    callable = Callable(definition, False)
+                    if callable.id not in self.referenced_usrs_:
+                        self.referenced_usrs_.append(callable.id)
+                    Index().register(callable)
 
     @staticmethod
     def _is_a_call(cursor):
```

Comment:
Is a method beginning with underscore a static method in Python?
if so, than having @staticmethod also is redundant.
I would prefer one or the other.

Reply:
A single leading underscore is a (weak "internal use" indicator)[https://www.python.org/dev/peps/pep-0008/] by convention.